test/hotspot/jtreg/vmTestbase/nsk/stress/strace/StraceBase.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package nsk.stress.strace;
+
+public class StraceBase {
+
+    private static final String[] EXPECTED_SYSTEM_CLASSES = {
+            "java.lang.ClassLoader",
+            "java.lang.System",
+            "java.lang.Object",
+            "java.lang.Thread",
+            "jdk.internal.event.ThreadSleepEvent",
+            "jdk.internal.misc.Blocker",
+            "jdk.internal.misc.VM",
+            "jdk.internal.vm.StackableScope",
+    };
+
+    private static boolean isValidJavaName(String name) {
+        if (!Character.isJavaIdentifierStart(name.charAt(0))) {
+            return false;
+        }
+        for (var ch : name.toCharArray()) {
+            if (!(Character.isJavaIdentifierPart(ch) || ch == '.')) {
+                return false;
+            }
+        }
+        return true;
+    }
+    /**
+     *   Method verifies that StackTraceElement is sane and might be expected it the current stack.
+     */
+    final static boolean checkElement(StackTraceElement element) {
+        String className = element.getClassName();
+        String methodName = element.getMethodName();
+        if (!isValidJavaName(className) || !isValidJavaName(methodName)) {
```

Comment:
You've pulled the class and method names out of a stacktrace element - how can they not be valid java names ???

Reply:
See my other comment. The tests doesn't always check names and methods so at least it is needed to verify that they are not some arbitrary string.

Reply:
But they have to be valid names barring some kind of native memory corruption, in which case you could be broken in a million different ways.

Reply:
I mot thinking of assigning some arbitrary string (ie name + signature instead of name). This is just a sanity check against such a mistake.

Reply:
I have updated verification and removed isValidJavaName() check,